src/taipy/core/config/checkers/_data_node_config_checker.py

```diff
@@ -22,7 +24,7 @@ def __init__(self, config: _Config, collector: IssueCollector):
         super().__init__(config, collector)
 
     def _check(self) -> IssueCollector:
-        data_node_configs = self._config._sections[DataNodeConfig.name]
+        data_node_configs: Dict[str, DataNodeConfig] = self._config._sections[DataNodeConfig.name]
         for data_node_config_id, data_node_config in data_node_configs.items():
```

Comment:
@jrobinAV 
This file has several failed mypy tests in line 61, 70, 78, 93, 94, 110 with errors related to Optional[Any] (I believe it's related to the variable data_node_config used in each function), can't explain the problem though. Can you have a look?

Reply:
Apparently, mypy detect that data_node_config.properties can be None. It is build from **properties in Section class in taipy-config repo. when no kwargs is passed, then properties is not an empty dict, but it is None. And so mypy tell us that some functions like "in" or "indexation" are incompatible with None type.

I just pushed a fix in taipy-config repo. It should be solved.

Reply:
@jrobinAV 
I have tried running again but the problem is still there. I also tried adding type Dict to the @property properties in Section with no help at all!

Reply:
I don't have it anymore. Are you sure you pulled the latest develop version of taipy-config?

Reply:
OK, I finally fixed the issue in the current PR as well. Can you try again?